plugins/parsers/avro/parser_test.go

```diff
@@ -0,0 +1,188 @@
+package avro
+
+import (
+	"bytes"
+	"encoding/binary"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/file"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/influxdata/toml"
+	"github.com/linkedin/goavro/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func jsonToAvroMessage(schemaID int32, schema string, message []byte) ([]byte, error) {
+	codec, err := goavro.NewCodec(schema)
+	if err != nil {
+		return nil, err
+	}
+
+	// We could use json.Unmarshal, but our codec can do it directly too.
+	native, _, err := codec.NativeFromTextual(message)
+	if err != nil {
+		return nil, err
+	}
+
+	binaryMsg, err := codec.BinaryFromNative(nil, native)
+	if err != nil {
+		return nil, err
+	}
+
+	// Put schemaID into []byte
+	schemaBytes := new(bytes.Buffer)
+	err = binary.Write(schemaBytes, binary.BigEndian, schemaID)
+	if err != nil {
+		return nil, err
+	}
+	schemaBin := schemaBytes.Bytes()
+
+	// Create serialized Avro binary message
+	magicByte := []byte{0x01}
+	binaryMsg = append(schemaBin, binaryMsg...)
+	binaryMsg = append(magicByte, binaryMsg...)
+
+	return binaryMsg, nil
+}
+
+type AvroCfg struct {
+	Inputs struct {
+		File []struct {
+			Parser
+			DataFormat string
+		}
+	}
+}
+
+func buildParser(buf []byte) (*Parser, error) {
+	var cfg AvroCfg
+
+	err := toml.Unmarshal(buf, &cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	pinput := cfg.Inputs.File[0].Parser
+	p := Parser{
+		MetricName:      pinput.MetricName,
+		SchemaRegistry:  pinput.SchemaRegistry,
+		Schema:          pinput.Schema,
+		Measurement:     pinput.Measurement,
+		Tags:            pinput.Tags,
+		Fields:          pinput.Fields,
+		Timestamp:       pinput.Timestamp,
+		TimestampFormat: pinput.TimestampFormat,
+		FieldSeparator:  pinput.FieldSeparator,
+		DefaultTags:     pinput.DefaultTags,
+	}
+	return &p, nil
+}
+
+func TestMultipleConfigs(t *testing.T) {
```

Comment:
I think your implementation of the tests is too complicated. Check the [binary parser test](https://github.com/influxdata/telegraf/blob/d1d9737da6f1dabd3350e5fdd80f7003ed207a81/plugins/parsers/binary/parser_test.go#L1407) for an example.
Basically there should be _no need at all_ for the `buildParser` function, nor for defining the input file. All of those values should be defined in the `telegraf.conf` file(s)!